lib/History.dart

```diff
@@ -0,0 +1,267 @@
+import 'package:flutter/material.dart';
+
+class History extends StatefulWidget {
+  @override
+  _HistoryState createState() => _HistoryState();
+}
+
+class _HistoryState extends State<History> {
```

Comment:
Since this page looks almost identical to the Upcoming rides page, simply re-use the widgets from there as much as possible to reduce the amount of code. If you fix the overflow issue in the Upcoming rides PR, it should be fixed here as well.